src/http/request_parser.rl

```diff
@@ -76,14 +76,13 @@ action checkpoint {
 }
 
 action assign_field {
-    if (_req->_headers.count(_field_name)) {
+    auto [iter, inserted] = _req->_headers.try_emplace(_field_name, std::move(_value));
+    if (!inserted) {
         // RFC 7230, section 3.2.2.  Field Parsing:
         // A recipient MAY combine multiple header fields with the same field name into one
         // "field-name: field-value" pair, without changing the semantics of the message,
         // by appending each subsequent field value to the combined field value in order, separated by a comma.
-        _req->_headers[_field_name] += sstring(",") + std::move(_value);
-    } else {
-        _req->_headers[_field_name] = std::move(_value);
+        iter->second += sstring(",") + std::move(_value);
```

Comment:
this is suspicious. at line 79, `_value` has been moved away. so the field would always be appended with a single "`,`", am i right? so i'd suggest just pass by a plain reference of `_value` at line 79.

Reply:
It's documented: "Unlike insert or emplace, these functions do not move from rvalue arguments if the insertion does not happen".

Reply:
but passing a rvalue reference to operator+ does not help, as `sstring::operator+(const sstring&)` is not able to consume its parameter -- it always performs a deep copy.

also, we might want to use the `operator+(const char(&s)[N], const sstring&)` by using something like
```c++
iter->second += "," + _value;
```

so we don't need to construct a temporary instance of `sstring` just to use `sstring::operator+(const sstring&)`.

Reply:
#1398 is supposed to ease the pain of formatting to sstring, so we don't need to worry about the order of parameters for better performance.

Reply:
`operator+` shouldn't be used where performance is important. It's minor here (since repeated header fields are rare) but can be improved.